src/network/protocol.rs

```diff
@@ -414,6 +435,149 @@ pub struct PortsRoot {
     pub ports: Vec<Port>,
 }
 
+/// ConntrackHelper.
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+pub struct ConntrackHelper {
+    pub helper: String,
+    pub protocol: String,
+    pub port: u16,
+}
+
+/// External gateway information.
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct ExternalGatewayInfo {
+    pub network_id: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub enable_snat: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub external_fixed_ips: Option<Vec<FixedIp>>,
+}
+
+/// A route.
+#[derive(Debug, Clone, Deserialize, Serialize, Copy, PartialEq)]
+pub struct Route {
```

Comment:
There is an identical `HostRoute` structure which I'd prefer to be reused.

Reply:
Perhaps we can make Route a type alias to HostRoute or vice versa.

Reply:
Yep, and deprecate `HostRoute`.